### ISSUE TITLE
💅 Consistent crash on malformed ternary

### ISSUE THREAD

**Issue by fmajestic** [biomejs/biome#1695]
### Environment information

```bash
CLI:
  Version:                      1.5.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v21.5.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "bun/1.0.25"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Rule name

`lint/correctness/noUnusedVariables`

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=dAB5AHAAZQAgAFQAbwBQAGEAcwBjAGEAbABDAGEAcwBlADwAUwAgAGUAeAB0AGUAbgBkAHMAIABzAHQAcgBpAG4AZwA%2BACAAPQAgAFMAIABlAHgAdABlAG4AZABzACAAYAAkAHsAaQBuAGYAZQByACAAUAByAGUAZgBpAHgAfQBgACAAPwAgADoAIAA7AA%3D%3D

### Expected result

I'm getting this crash message:
```
Source Location: /home/runner/work/biome/biome/crates/biome_js_semantic/src/semantic_model/model.rs:334:46
Thread Name: main
Message: no entry found for key
```

Just as the placeholder for this field says, "It should not throw an error." 😝

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
The code that you pasted is not valid. Look at the playground, and even Prettier throws an error. [And TS too](https://www.typescriptlang.org/play?#code/C4TwDgpgBAKg9gBQIYGcDGSA2BhVEA8AylBAB7AQB2AJilCsAE4CWlA5gHxQC8UxZFGnQAGAEgDerAGYRGUBIwhTmpAL7CoAfigAuKAG4gA)

**Comment by fmajestic** [comment]
Yeah, that's why I specifically said _malformed_ ternary :D. `const [xyy:?:""\n} = bar;` is also invalid code but doesn't cause a hard crash.

I managed to narrow down the crash to that specific lint rule, but it happened while I was editing a file and typed in an "empty" ternary like that.

**Comment by ematipico** [comment]
Yeah, but the issue is not the linter. It's a parser issue :)

**Comment by togami2864** [comment]
I'll take a look:)

**Comment by togami2864** [comment]
The parser seems to be working fine, but the issue only reproduces in the production environment🤔

https://github.com/biomejs/biome/assets/62130798/c0557b37-d0dc-45e4-8710-82e734779b75

**Comment by ah-yu** [comment]
Hey @togami2864 , I took a look at this issue before and have some observations that I hope will be helpful:
- This issue only happens when the `NoUnusedVariable` rule is enabled. 
- This issue disappears when the true branch of the condition type is not empty [playground](https://biomejs.dev/playground/?lintRules=all&code=dAB5AHAAZQAgAEYAbwBvADwAVAA%2BACAAPQAgAFQAIABlAHgAdABlAG4AZABzACAAaQBuAGYAZQByACAAVAAgAD8AIAAgADEAOgAgAA%3D%3D)

I guess you can explore the issue more from here
https://github.com/biomejs/biome/blob/ee9b3ac6c7338f7718bf685159daf90e4396b583/crates/biome_js_semantic/src/events.rs#L484-L487

**Comment by togami2864** [comment]
@ah-yu Awesome! Thank you!

**Comment by Conaclos** [comment]
@togami2864 Are you still working on it? If not, I could take a look :)

**Comment by togami2864** [comment]
How quickly did you fix it 🎉 Thank you!
